### PR TITLE
Added a raw data API

### PR DESCRIPTION
**Pull request by vaxerski**
See `getRawShapeData` (C++) or `hyprcursor_get_raw_shape_data` (C)

ref #20

cc @jinliu for KDE
cc @fufexan

How does this look for you?

### CODE REVIEW

**Comment by vaxerski** [comment]
also @fufexan can you fix ci for once xd

**Comment by fufexan** [comment]
Should work if you rebase on main.

**Comment by vaxerski** [comment]
thanks

**Comment by jinliu** [comment]
Looks good to me.

**Comment by vaxerski** [comment]
alright. Thanks! I'll clean up some docs then tomorrow and merge. Or, well, later today. It's 3am.

**Comment by jinliu** [comment]
Thanks! And please take your time, as quite a lot need to be done in the KDE side so no hurry.

**Comment by vaxerski** [comment]
Now the real dilemma, what will come first, hyprcursors in kde or me getting banned from freedesktop :P

File: include/hyprcursor/shared.h [hunk 1]
```diff
@@ -25,6 +25,39 @@ enum eHyprcursorLogLevel {
...
+    int               size;
+    int               delay;
+};
+
+typedef struct SCursorRawShapeImageC hyprcursor_cursor_raw_shape_image;
+
+struct SCursorRawShapeDataC {
+    struct SCursorRawShapeImageC* images;
+    unsigned long int             len;
+    float                         hotspotX;
```
[the remark this answers is not in the document]
Comment (jinliu) [reply include/hyprcursor/shared.h:52]: Nevermind. Just realized that in SCursorShapeData hotspots are coords in the pixmap, so they must be per-pixmap.

[the remark this answers is not in the document]
Comment (vaxerski) [reply include/hyprcursor/shared.h:52]: yea here it's just the config one as float 0..1

multiply that by your pixel size and you get your pixel hotspot

**Pull request merged**